src/spark/state.cpp

```diff
@@ -583,10 +637,6 @@ bool CheckSparkSpendTransaction(
 
     LogPrintf("CheckSparkSpendTransaction: tx metadata hash=%s\n", txHashForMetadata.ToString());
 
-    if (!fStatefulSigmaCheck) {
```

Comment:
why do we need to move this check to lower lines?

Reply:
okay I see your point, so you verify the proof even it is not stateful check in a separate thread,
so in case it is mempool verification it is stateful, checkblock and connectblock are running this function one with stateful check one without, so what is the purpose  makeing the code more complex by adding mmore threads and so on?